maitake-sync/src/wait_cell.rs

```diff
@@ -327,18 +327,70 @@ impl WaitCell {
         }
     }
 
-    /// Asynchronously poll the [`WaitCell`] until a condition occurs
+    /// Asynchronously poll the given function `f` until a condition occurs,
+    /// using the [`WaitCell`] to only re-poll when notified.
     ///
     /// This can be used to implement a "wait loop", turning a "try" function
     /// (e.g. "try_recv" or "try_send") into an asynchronous function (e.g.
     /// "recv" or "send").
     ///
+    /// In particular, this function correctly *registers* interest in the [`WaitCell`]
```

Comment:
very tiny nit: maybe worth linking to the `subscribe` method here where you say "registers"?

Reply:
I could, but it's through the `Wait` struct, and I didn't want to be distracting. If you want it, I can add it.

Reply:
not a big deal to me either way, this is fine!